src/Platforms/Echo.Platforms.AsmResolver/Emulation/CliMarshaller.cs

```diff
@@ -28,18 +28,21 @@ private TypeSignature GetElementType(TypeSignature type)
         {
             switch (type.ElementType)
             {
-                case ElementType.Enum:
+                case ElementType.ValueType:
                     if (!_resolvedTypes.TryGetValue(type, out var definition))
                     {
                         definition = type.Resolve();
                         if (definition is not null)
                             _resolvedTypes.Add(type, definition);
                     }
 
-                    if (definition?.GetEnumUnderlyingType() is not { } enumUnderlyingType)
+                    if (!definition?.IsEnum ?? true)
```

Comment:
We probably want to replace this with `if (definition is null || !definition.IsEnum)`  for both readability and to avoid the NRT warnings.